react-web/src/hooks/useLogs.ts

```diff
@@ -13,41 +14,119 @@ export const useLogs = (
     testEnded: boolean,
     handleErrorScenario: () => void
 ) => {
+    const dispatch = useDispatch()
     const [logInfo, setLogInfo] = useState<Log[]>([])
     const [fetchingLogs, setFetchingLogs] = useState(false);
     const [refetchLogsOffset] = useState(1);
+    const [enabled, setEnabled] = useState(false)
+
+    const { startTime, endTime, runState, ended } = useAppSelector((state) => state.runTime);
+
+    useEffect(() => {
+        if (testEnded) { 
+            dispatch(setEnded(ended+1)); 
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [testEnded])
+
+
+    useEffect(() => {
+        setEnabled(!fetchingLogs && !(ended > 1) && !!uuid)
+    }, [fetchingLogs, ended, uuid])
```

Comment:
unnecessarily and possibly harmful way of using `useEffect` as a state changer.
https://beta.reactjs.org/learn/you-might-not-need-an-effect

```suggestion
   const enabled = !fetchingLogs && !(ended > 1) && !!uuid
```

Reply:
I do understand this suggestion, and it was in a const variable before moving it into state var in this PR. I did this, as part of performance optimization to minimize recalculations. I will recheck this and get back soon.

Reply:
@bogdan-manole I've done this change, pls re-review